### DISCUSSION TITLE
How to create a list of publications?

### DISCUSSION THREAD

**Discussion by edudimitrov** [quarto-dev/quarto-cli#2925] [Q&A]
I was wondering the easiest way to get a list of publications from Google Scholar.
Here is what I tried:
1. I used [PyPaperBot](https://github.com/ferru97/PyPaperBot) to download pdf files and a csv
2. I converted the csv to a yml file
3. I tried to use the [yml file](https://github.com/edudimitrov/edudimitrov.github.io/blob/main/publica.yml) as a [listing](https://github.com/edudimitrov/edudimitrov.github.io/blob/main/publicacoes.qmd)

But the listing didn't work.  I just wanted a list of publications and links to the pdf files.  Similar to [this example](https://www.mm218.dev/papers/).

PyPaperBot generates a bibtex file, so if there was a way to generate a list of publications from a bibtex file, that would also solve the matter.

**Comment by mcanouil** [comment]
The following is a WIP listing for publication based on a bibtex file: https://github.com/mcanouil/mickael.canouil.fr/blob/fix-issue23/publications.qmd
It uses `pandoc file.bib --standalone --from=bibtex --to=markdown` to create the base YAML that I modify for my needs.

Another example of custom listing for publication by @jmgirard: https://github.com/jmgirard/affcomlab

**Reply by sjkiss** [reply]
Hi there , your link is no longer working. Could I take a look at it?

**Reply by mcanouil** [reply]
- https://github.com/mcanouil/mickael.canouil.fr/blob/main/publications.qmd

**Comment by dfolio** [comment]
I‌ have proposed a brief tutorial on how to use [Quarto listing](https://quarto.org/docs/websites/website-listings.html "Documentation about document Listings") to create a publication list, see it here: <https://dfolio.fr/posts/2023/2023-08-17_publications.html> and see the result here <https://dfolio.fr/publications/>. This method is "_quite_" simple, but may be tedious if you have a lot of publications.

Another simple solution, is to use [Pandoc](https://pandoc.org/) to render citations with [citeproc](https://pandoc.org/MANUAL.html#citations). Put all references in a bibliographic data file (e.g., a `.bib` file), include `nocite: @*` in the metadata front end, and add [div block with id `refs`](https://pandoc.org/MANUAL.html#placement-of-the-bibliography) in your references documents.\
One disadvantage of this approach is that the references are not clearly distinguished or organized.

There are other methods for managing bibliography references, particularly if you already use [BibLaTeX](https://ctan.org/pkg/biblatex) or [BibTeX](https://ctan.org/pkg/bibtex). In my case, I use [RefManageR](https://docs.ropensci.org/RefManageR/), and some brief explaination are discussed on this page <https://dfolio.fr/posts/2024/2024-04-29_refmanager.html>, and see my reference page: <https://dfolio.fr/research/references.html>

I hope that this will prove useful in the creation of your own publication list.

**Comment by drganghe** [comment]
Here is an example of autogenerate a publications list from BIB converted Excel file while keeping the category, filtering functionality of Quarto Listing.

`xlsx_to_yml.py`

`publications.yml` + `pub-listing.ejs` + `pub-listing.css`

https://drganghe.github.io/notes/generate-publications-listing.html
